jax/_src/custom_derivatives.py

```diff
@@ -298,119 +311,31 @@ def _apply_todos(todos, outs):
 
 custom_jvp_call_p = CustomJVPCallPrimitive('custom_jvp_call')
 
+def _custom_jvp_call_typecheck(*in_avals, call_jaxpr, jvp_jaxpr_thunk, num_consts):
+  # TODO(mattjj): could do more checking here...
```

Comment:
Why prevent `core.check_call`? If it's necessary, then can we still carry out some of the checks from `core.check_call`, if not all?

Reply:
`check_call` doesn't quite work for `custom_jvp_call`; for example, there's no `call_jaxpr` parameter here. Moreover there are other things we might want to check here, like `num_consts` and maybe some things about `jvp_jaxpr_thunk` too.

So while I suspect we can factor out some shared helper functions, in general `custom_jvp_call` will need a custom typecheck rule.